tensorflow/lite/tools/benchmark/experimental/firebase/android/jni/benchmark_model_jni.cc

```diff
@@ -253,11 +203,25 @@ extern "C" {
 
 JNIEXPORT void JNICALL
 Java_org_tensorflow_lite_benchmark_firebase_BenchmarkModel_nativeRun(
-    JNIEnv* env, jclass clazz, jstring library_dir, jint scenario,
+    JNIEnv* env, jclass clazz, jstring library_dir, jstring args_obj,
     jint report_fd) {
   const char* lib_dir = env->GetStringUTFChars(library_dir, nullptr);
+  const char* args_chars = env->GetStringUTFChars(args_obj, nullptr);
+
+    // Split the args string into individual arg tokens.
+    std::istringstream iss(args_chars);
+    std::vector<std::string> args_split{std::istream_iterator<std::string>(iss),
+                                        {}};
+
+    // Construct a fake argv command-line object for the benchmark.
+    std::vector<char*> argv;
+    std::string arg0 = "(BenchmarkModelAndroid)";
+    argv.push_back(const_cast<char*>(arg0.data()));
+    for (auto& arg : args_split) {
+      argv.push_back(const_cast<char*>(arg.data()));
+    }
```

Comment:
Condolences... for having to see that. Yes, that's C++. 
No matter how hard they tie El-Sid's dead body to a horse, he was very dead.

I loved
```c++
for (auto& arg... 
```
Gosh, I had a crash on that language _(and so many literal crashes due to that language)_ in my late 20's...but now I just think it's funny how they try to keep it alive.